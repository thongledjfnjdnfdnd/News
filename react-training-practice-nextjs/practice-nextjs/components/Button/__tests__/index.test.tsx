import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Button, ButtonProps } from '@components';
import { fireEvent, render } from '@testing-library/react';

describe('Test [Button] component', () => {
  const handleOnClickButton = jest.fn();
  const initButton = {
    onClick: handleOnClickButton,
    children: 'Create New Meetup',
    testId: 'test-button-id',
    className: 'shadow',
  };

  test('Component [Button] should render correctly with default value Props', () => {
    const component = renderer
      .create(<Button {...({} as ButtonProps)}></Button>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [Button] should render correctly', () => {
    const component = renderer
      .create(<Button typeButton="button" {...initButton} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [Button] call function handleOnClickButton when user click button', () => {
    const { getByTestId } = render(
      <Button typeButton="button" {...initButton} />,
    );
    const button = getByTestId('test-button-id');
    expect(button.textContent).toEqual('Create New Meetup');
    expect(button).toHaveProperty('type', 'button');
    fireEvent.click(button);
    expect(handleOnClickButton).toHaveBeenCalled();
  });
});
