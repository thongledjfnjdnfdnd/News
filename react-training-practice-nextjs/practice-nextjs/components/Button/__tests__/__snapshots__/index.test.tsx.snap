// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [Button] component Component [Button] should render correctly 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-400 text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded shadow"
  data-testid="test-button-id"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Create New Meetup
</button>
`;

exports[`Test [Button] component Component [Button] should render correctly with default value Props 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-400 text-white font-bold py-2 px-4 border-b-4 border-blue-700 hover:border-blue-500 rounded "
  disabled={false}
  type="button"
/>
`;
